llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp

```diff
@@ -13003,29 +13047,6 @@ unsigned ARMAsmParser::validateTargetOperandClass(MCParsedAsmOperand &AsmOp,
     if (hasV8Ops() && Op.isReg() && Op.getReg() == ARM::SP)
       return Match_Success;
     return Match_rGPR;
-  // Note: This mutates the operand which could cause issues for future
-  // matches if this one fails later.
```

Comment:
Yuck! Thanks for removing this too.

Reply:
Admittedly I am responsible for this so can't take too much credit